File: May 2, 2026.py
def get_deepest_brackets(s):

    current = ""
    result = ""
    count = 0
    max_count = 0

    for char in s:
        if char == "(" or char == "[" or char == "{":
            count += 1
            if count > max_count:
                max_count = count
                current = ""
            continue
        
        if char == ")" or char == "]" or char == "}":
            if count == max_count:
                result = current
            count -= 1
            continue
        
        if count == max_count:
            current += char

    return result

File: May 2, 2026.js
function getDeepestBrackets(str) {

  let current = "";
  let result = "";
  let count = 0;
  let maxCount = 0;

  for (let i = 0; i < str.length; i++) {
    const char = str[i];

    if (char == "(" || char == "[" || char == "{") {
      count += 1;
      if (count > maxCount) {
        maxCount = count;
        current = "";
      }
      continue;
    }

    if (char == ")" || char == "]" || char == "}") {
      if (count == maxCount) {
        result = current;
      }
      count -= 1;
      continue;
    }

    if (count == maxCount) {
      current += char;
    }

  }

  return result;
}